DatadogLogs/Sources/Feature/MessageReceivers.swift

```diff
@@ -277,56 +274,45 @@ internal struct WebViewLogReceiver: FeatureMessageReceiver {
     ///   - message: The Feature message
     ///   - core: The core from which the message is transmitted.
     func receive(message: FeatureMessage, from core: DatadogCoreProtocol) -> Bool {
-        do {
-            guard case let .baggage(label, baggage) = message, label == LoggingMessageKeys.browserLog else {
-                return false
-            }
-
-            guard var event = try baggage.encode() as? [String: Any?] else {
-                throw InternalError(description: "event is not a dictionary")
-            }
+        guard case var .log(event) = message.value(WebViewMessage.self) else {
+            return false
+        }
 
-            let versionKey = LogEventEncoder.StaticCodingKeys.applicationVersion.rawValue
-            let envKey = LogEventEncoder.StaticCodingKeys.environment.rawValue
-            let tagsKey = LogEventEncoder.StaticCodingKeys.tags.rawValue
-            let dateKey = LogEventEncoder.StaticCodingKeys.date.rawValue
+        let versionKey = LogEventEncoder.StaticCodingKeys.applicationVersion.rawValue
+        let envKey = LogEventEncoder.StaticCodingKeys.environment.rawValue
+        let tagsKey = LogEventEncoder.StaticCodingKeys.tags.rawValue
+        let dateKey = LogEventEncoder.StaticCodingKeys.date.rawValue
 
-            core.scope(for: LogsFeature.name)?.eventWriteContext { context, writer in
-                let ddTags = "\(versionKey):\(context.version),\(envKey):\(context.env)"
+        core.scope(for: LogsFeature.name)?.eventWriteContext { context, writer in
+            let ddTags = "\(versionKey):\(context.version),\(envKey):\(context.env)"
 
-                if let tags = event[tagsKey] as? String, !tags.isEmpty {
-                    event[tagsKey] = "\(ddTags),\(tags)"
-                } else {
-                    event[tagsKey] = ddTags
-                }
+            if let tags = event[tagsKey] as? String, !tags.isEmpty {
+                event[tagsKey] = "\(ddTags),\(tags)"
+            } else {
+                event[tagsKey] = ddTags
+            }
 
-                if let timestampInMs = event[dateKey] as? Int64 {
-                    let serverTimeOffsetInMs = context.serverTimeOffset.toInt64Milliseconds
-                    let correctedTimestamp = Int64(timestampInMs) + serverTimeOffsetInMs
-                    event[dateKey] = correctedTimestamp
-                }
+            if let timestampInMs = event[dateKey] as? Int {
```

Comment:
The `date` type is now `Int` as it's not going through the encoding/decoding of a baggage, the process resulted in `Int64`.